File: app/src/main/java/com/robotnec/budget/core/service/aggregation/AggregationService.java
package com.robotnec.budget.core.service.aggregation;

import com.robotnec.budget.core.domain.operation.MoneyOperation;

import java.util.List;

/**
 * @author zak <zak@swingpulse.com>
 */
public interface AggregationService {

    TransactionAggregation aggregate(List<MoneyOperation> transactions, Resolution resolution);

    enum Resolution {
        DAY
    }
}

File: app/src/main/java/com/robotnec/budget/core/service/aggregation/AggregationService.kt
package com.robotnec.budget.core.service.aggregation

import com.robotnec.budget.core.domain.operation.Transaction
import com.robotnec.budget.core.service.aggregation.impl.TransactionAggregation

/**
 * @author zak zak@swingpulse.com>
 */
interface AggregationService {

    fun aggregate(transactions: List<Transaction>, resolution: Resolution): TransactionAggregation

    enum class Resolution {
        DAY
    }
}
